tools/osv-scanner/internal/osv/osv.go

```diff
@@ -40,11 +43,15 @@ type BatchedQuery struct {
 	Queries []*Query `json:"queries"`
 }
 
+// Vulnerability represents a vulnerability entry from OSV.
+type Vulnerability struct {
```

Comment:
Prefer having the batched vulnerability and the hydrated version be two different types. This way it's clear that the Group  only takes already hydrated elements.

Reply:
Good suggestion, done!